src/ucp/wireup/select.c

```diff
@@ -1426,29 +1427,94 @@ ucp_wireup_is_md_map_count_valid(ucp_context_h context, ucp_md_map_t md_map)
            (ucs_popcount(md_map) < UCP_MAX_OP_MDS);
 }
 
+static double ucp_wireup_get_lane_progress_overhead(ucp_worker_h worker,
+                                                    ucp_rsc_index_t rsc_index)
+{
+    ucp_worker_iface_t *wiface = ucp_worker_iface(worker, rsc_index);
+    uct_perf_attr_t perf_attr;
+
+    perf_attr.field_mask = UCT_PERF_ATTR_FIELD_OPERATION |
+                           UCT_PERF_ATTR_FIELD_PROGRESS_OVERHEAD;
+
+    /* We use AM_SHORT as a dummy operation because we don't know which
+        protocol will be selected (anyway it has no effect on the value of
+        progress_overhead). */
+    perf_attr.operation = UCT_EP_OP_AM_SHORT;
+
+    if (uct_iface_estimate_perf(wiface->iface, &perf_attr) != UCS_OK) {
+        /* Skip the overhead check if no value was retrieved */
+        return 0;
+    }
+
+    return perf_attr.progress_overhead;
+}
+
+static unsigned
+ucp_wireup_add_fast_lanes(const ucp_wireup_select_params_t *select_params,
+                          const ucp_wireup_select_info_t *sinfo_array,
+                          unsigned num_sinfo, ucp_worker_h worker,
+                          ucp_lane_type_t lane_type,
+                          ucp_wireup_select_context_t *select_ctx)
+{
+    unsigned num_lanes     = 0;
+    double min_overhead    = DBL_MAX;
+    ucp_context_h context  = worker->context;
+    const double max_ratio = 1. / context->config.ext.multi_lane_max_ratio;
```

Comment:
now the same parameter is used as ratio of BW and progress overhead, which is confusing

Reply:
maybe we should rename it to some more general purpose parameter?